drivers/clock_control/clock_stm32_ll_h5.c

```diff
@@ -434,6 +434,17 @@ static int set_up_plls(void)
 #endif
 
 #if defined(STM32_PLL_ENABLED)
+
+#if defined(CONFIG_STM32_MEMMAP) && defined(CONFIG_BOOTLOADER_MCUBOOT)
+	/*
+	 * If MEMMAP and MCUBOOT, then the OCTOSPI is necessarily also defined,
+	 * don't disable PLL during application initialization that runs in memmap mode
+	 * when (Q/O)SPI uses PLL as its clock source but exit.
+	 */
+	if (LL_RCC_GetOSPIClockSource(LL_RCC_OSPI_CLKSOURCE) == LL_RCC_OSPI_CLKSOURCE_PLL1Q) {
+		return 0;
```

Comment:
I think returning now is not awaited. Maybe PLL2 and PLL3 are also to be enabled.
Implementation would maybe be simpler with 3 local helper functions: `set_up_pll1()`, `set_up_pll2()` and `set_up_pll3()`.